Scissors  - Ivana

i-vs opened the pull request.

i-vs committed:
Intial commit

i-vs committed:
Refactor multiple functions and add a new helper function

beccaelenzil commented on viewing_party/main.py:
```diff
@@ -0,0 +1,271 @@
+# test_wave_01
+
+def create_movie(movie_title=str, genre=str, rating=float):
+    """Creates a ``new_movie`` that will be added to the user's list.
```
Nice use of docstrings throughout

beccaelenzil commented on viewing_party/main.py:
```diff
@@ -0,0 +1,271 @@
+    default_rating = 0.0
+    avg_rating = 0.0
+
+    if movies_list_length > 0:
+        for movie in user_movies:
+            avg_rating += movie["rating"]
+    else:
+        return default_rating
```
You might consider dealing with returning the default rating first to deal with the edge case first, such as:
```suggestion
    if movies_list_length == 0:
        return None
        
   for movie in user_movies:
        avg_rating += movie["rating"]
```

beccaelenzil commented on viewing_party/main.py:
```diff
@@ -0,0 +1,271 @@
+        if genre not in all_genres:
+            all_genres[genre] = 1
+        else:
+            all_genres[genre] = all_genres[genre] + 1  # update the value
```
Consider using `+=`

```suggestion
            all_genres[genre] += 1  # update the value
```

beccaelenzil commented on viewing_party/main.py:
```diff
@@ -0,0 +1,271 @@
+
+# NEW: Helper function: gets the movies all friends watched.
+
+def get_friends_movies(user_data):
```
Great use of a helper function to DRY up your code!

beccaelenzil commented on viewing_party/main.py:
```diff
@@ -0,0 +1,271 @@
+            the user's favorite genre.
+    """
+
+    friends_movies = get_friends_movies(user_data)
+    fav_genre = get_most_watched_genre(user_data)
```
Great work using the previous functions as helper function to write clear, concise code

beccaelenzil reviewed (commented):
Great work on your first Ada project! You've met the learning goals of working with loops, conditionals, and nested data structures. Your code is clear and concise, and makes good use of helper functions. I've made a couple suggestions that we can discuss in your 1:1 this week. Keep up the hard work!

GREEN

i-vs committed:
Update viewing_party/main.py

Return the default rating first to deal with the edge case first to improve efficiency

Co-authored-by: Becca  <becca@adadevelopersacademy.org>

i-vs committed:
Update viewing_party/main.py

Change the assignment operator for better readability

Co-authored-by: Becca  <becca@adadevelopersacademy.org>
